Project/Source/StateMachineManager.cpp

```diff
@@ -17,54 +16,59 @@ bool StateMachineManager::Contains(std::list<AnimationInterpolation>& animationI
 	return false;
 }
 
-void StateMachineManager::SendTrigger(const std::string& trigger, std::unordered_map<UID, float>& currentTimeStates, std::list<AnimationInterpolation>& animationInterpolations, const UID& stateMachineResourceUID, State& currentState, State& currentStateSecondary, std::unordered_map<UID, float>& currentTimeStatesPrincipal, StateMachineEnum stateMachineEnum, std::list<AnimationInterpolation>& animationInterpolationsSecondary) {
+void StateMachineManager::SendTrigger(const std::string& trigger, StateMachineEnum stateMachineSelected, ComponentAnimation& componentAnimation) {
+	bool isPrincipal = stateMachineSelected == StateMachineEnum::PRINCIPAL;
+	UID stateMachineResourceUID = isPrincipal ? componentAnimation.stateMachineResourceUIDPrincipal : componentAnimation.stateMachineResourceUIDSecondary;
+	State* currentState = isPrincipal ? &componentAnimation.currentStatePrincipal : &componentAnimation.currentStateSecondary;
+	std::unordered_map<UID, float>* currentTimeStates = isPrincipal ? &componentAnimation.currentTimeStatesPrincipal : &componentAnimation.currentTimeStatesSecondary;
+	std::list<AnimationInterpolation>* animationInterpolations = isPrincipal ? &componentAnimation.animationInterpolationsPrincipal : &componentAnimation.animationInterpolationsSecondary;
+
 	ResourceStateMachine* resourceStateMachine = App->resources->GetResource<ResourceStateMachine>(stateMachineResourceUID);
 	if (!resourceStateMachine) {
 		return;
 	}
 	Transition* transition = resourceStateMachine->FindTransitionGivenName(trigger);
 	if (transition != nullptr) {
-		if (transition->source.id == currentState.id) {
-			currentTimeStates[transition->target.id] = 0;
+		if (transition->source.id == (*currentState).id) {
```

Comment:
This is not something you have to change, but 'currentState->id' would be equivalent to '(*currentState).id'.